### PR TITLE
Refine errors to remove changes breaking design

### PR DESCRIPTION
**Pull request by yah01**
/kind improvement

### CODE REVIEW

File: internal/proxy/lb_policy.go [hunk 1]
```diff
@@ -167,7 +168,7 @@ func (lb *LBPolicyImpl) ExecuteWithRetry(ctx context.Context, workload ChannelWo
 
 			// cancel work load which assign to the target node
 			lb.balancer.CancelWorkload(targetNode, workload.nq)
-			return merr.WrapErrShardDelegatorAccessFailed(workload.channel, err.Error())
+			return errors.Wrapf(err, "failed to get delegator %d for channel %s", targetNode, workload.channel)
```
Comment (jiaoew1991) [inline internal/proxy/lb_policy.go:171]: We need guidelines on when to use "errors" and when to use "merr".

**Comment by yah01** [comment]
rerun ut

**Pull request merged**